src/libraries/System.IO.FileSystem/tests/Base/BaseGetSetTimes.cs

```diff
@@ -54,7 +57,7 @@ public static TimeFunction Create(SetTime setter, GetTime getter, DateTimeKind k
             public DateTimeKind Kind => Item3;
         }
 
-        private void SettingUpdatesPropertiesCore(T item)
+        protected virtual void SettingUpdatesPropertiesCore(T item, bool isLink)
```

Comment:
Can this go back to being non-virtual?
And private?

Can you default isLink to `false`?

Reply:
> Can you default isLink to false?

Or better. Remove the `if (!isLink || SettingPropertiesUpdatesLink)` condition and do something like:

```
private void SettingUpdatesPropertiesCore(T item, T? linkTarget = null)
{
   bool isLink = linkTarget is not null;
...
    DateTime dt = new DateTime(2014, 12, 1, 12, 3, 3, LowTemporalResolution ? 0 : 321, function.Kind);
   function.Setter(item, dt);
   T target = !isLink || SettingPropertiesUpdatesLink ? item : linkTarget;
   DateTime result = function.Getter(target);
}
```

Reply:
Can you also add a `ToString` to `TimeFunction` which makes it easier to understand what items failed?

Something like:
```cs
            public override string ToString()
            {
                return $"TimeFunction DateTimeKind.{Kind} Setter: {Setter.Method.Name} Getter: {Getter.Method.Name}";
            }
```